src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ChannelService.java

```diff
@@ -0,0 +1,224 @@
+        savedChannel.getConversationParticipants().add(conversationParticipantOfRequestingUser);
+        savedChannel = channelRepository.save(savedChannel);
+        conversationService.broadcastOnConversationMembershipChannel(course, MetisCrudAction.CREATE, savedChannel, Set.of(user));
+        return savedChannel;
+    }
+
+    /**
+     * Checks if the given channel is valid for the given course or throws an exception
+     *
+     * @param courseId the id of the course
+     * @param channel  the channel to check
+     */
+    public void channelIsValidOrThrow(Long courseId, Channel channel) {
+        var validator = buildDefaultValidatorFactory().getValidator();
+        var violations = validator.validate(channel);
+        if (!violations.isEmpty()) {
+            throw new ConstraintViolationException(violations);
+        }
+        if (!StringUtils.hasText(channel.getName())) {
+            throw new BadRequestAlertException("A channel needs to have a name", CHANNEL_ENTITY_NAME, "nameMissing");
+        }
+        if (!channel.getName().matches(CHANNEL_NAME_REGEX)) {
+            throw new BadRequestAlertException("Channel names can only contain lowercase letters, numbers, and dashes.", CHANNEL_ENTITY_NAME, "namePatternInvalid");
+        }
+        Optional<Channel> channelWithSameName;
+        if (channel.getId() != null) {
+            channelWithSameName = channelRepository.findChannelByCourseIdAndNameAndIdNot(courseId, channel.getName(), channel.getId());
+        }
+        else {
+            channelWithSameName = channelRepository.findChannelByCourseIdAndName(courseId, channel.getName());
+        }
+        channelWithSameName.ifPresent(existingChannel -> {
+            throw new ChannelNameDuplicateException(existingChannel.getName());
+        });
+    }
+
+    /**
+     * Archive the channel with the given id
+     *
+     * @param channelId the id of the channel to archive
+     */
+    public void archiveChannel(Long channelId) {
+        var channel = getChannelOrThrow(channelId);
+        if (channel.getIsArchived()) {
+            return;
+        }
+        channel.setIsArchived(true);
+        var updatedChannel = channelRepository.save(channel);
+        conversationService.notifyAllConversationMembersAboutUpdate(updatedChannel);
+    }
+
+    /**
+     * Unarchive the channel with the given id
+     *
+     * @param channelId the id of the archived channel to unarchive
+     */
+    public void unarchiveChannel(Long channelId) {
+        var channel = getChannelOrThrow(channelId);
+        if (!channel.getIsArchived()) {
+            return;
+        }
+        channel.setIsArchived(false);
+        var updatedChannel = channelRepository.save(channel);
+        conversationService.notifyAllConversationMembersAboutUpdate(updatedChannel);
+    }
+
+    /**
+     * Delete the channel with the given id
+     *
+     * @param channel the channel to delete
+     */
+    public void deleteChannel(Channel channel) {
```

Comment:
Call the repository method directly?

Reply:
I do not like to break the Resource -> Service -> Repository layer that often tbh. I think its not "bad" to have it in the service. So I do not need to add a new dependency to the resource